src/features/defi/providers/cosmos/components/CosmosManager/Overview/CosmosOverview.tsx

```diff
@@ -39,81 +38,94 @@ type CosmosOverviewProps = {
 }
 
 export const CosmosOverview: React.FC<CosmosOverviewProps> = ({
-  accountId: defaultAccountId,
+  accountId,
   onAccountIdChange: handleAccountIdChange,
 }) => {
   const translate = useTranslate()
   const { query, history, location } = useBrowserRouter<DefiQueryParams, DefiParams>()
-  const { defaultAccountId: queryAccountId, chainId, contractAddress, assetReference } = query
-
-  const accountId = useMemo(
-    () => defaultAccountId ?? queryAccountId,
-    [defaultAccountId, queryAccountId],
-  )
+  const {
+    // defaultAccountId: queryAccountId, // TODO: do we even need this
```

Comment:
This one is a super hacky hack from previous implementations and was propagated where it shouldn't. Will investigate blames some more and most likely remove it